Deploy/prod

rxjeong opened the pull request:
Pull Request Deploy/prod to Main branch

mishraomp commented on configurations.ini:
```diff
@@ -0,0 +1,30 @@
+AUTH_URL = https://dev.oidc.gov.bc.ca
+# AUTH_URL = https://test.oidc.gov.bc.ca # test
+# AUTH_URL	= https://oidc.gov.bc.ca # prod
+CHEFS_URL = https://ches-dev.apps.silver.devops.gov.bc.ca
```
This is CHES URL, could you please rename the variable? Also could we please update the URL to not use `DEV` environment of CHES.
Some more best practices on CHES https://github.com/bcgov/common-hosted-email-service/wiki/Best-Practices

rxjeong replied:
1.Renamed CHEFS_URL to CHES_URL
2.Updated to use PROD
3.Updated to use "donotreplySRIS@gov.bc.ca" for sender address. (I wonder if we can use the address)
Am I missing here?

mishraomp replied:
I think the best option for from address is we add to env variable, so that if change is required, code changes wont be required.

rxjeong replied:
The MAPHUB_URL and WEBMAP_POPUP_URL in the configurations.ini file would be unlikely to be changed, but the CSV/Layer IDs and CHES and CHEFS URL could be changed. I will remove them from configuration file and add them into the GitHub setting.

mishraomp commented on configurations.ini:
```diff
@@ -0,0 +1,30 @@
+WEB_MAP_APP_ID = 8a6afeae8fdd4960a0ea0df1fa34aa74
+[CHEFS]
+CHEFS_API_URL = https://submit.digital.gov.bc.ca/app/api/v1
+AUTH_URL = https://dev.oidc.gov.bc.ca
```
could you please update the URL to use `PROD` environment.

rxjeong replied:
Changed AUTH_URL to use PROD

mishraomp commented on helper.py:
```diff
@@ -0,0 +1,175 @@
+    _lat_dd = 0
+    _lon_dd = 0
+    try:
+        EXP_EXTRACT_FLOATING = r'[-+]?\d*\.\d+|\d+'
```
could you please make this lowercase, I believe making variables inside a function lowercase makes it distinctive from global variables.

rxjeong replied:
The "EXP_EXTRACT_FLOATING" variable is used from other functions, so it was moved to constants.py and it keeps uppercase.

mishraomp commented on helper.py:
```diff
@@ -0,0 +1,175 @@
+            if _lon_dd > 0: _lon_dd = - _lon_dd # longitude degrees should be minus in BC bouding box
+
+    except ValueError as ve:
+        print('ValueError Raised:', ve)
```
this string `'ValueError Raised:'` is repeated multiple times, could you please define a constant?

rxjeong replied:
Defined as a constant in the constant.py

mishraomp commented on helper.py:
```diff
@@ -0,0 +1,175 @@
+  ches_response = None
+  try:
+    auth_pay_load = 'grant_type=client_credentials'
+    auth_haders = {
```
A small typo, could you please fix it.

rxjeong replied:
fixed the auth_haders -> 'auth_headers'

mishraomp commented on helper.py:
```diff
@@ -0,0 +1,175 @@
+      'Authorization': 'Bearer ' + access_token
+      }
+      ches_response = requests.request("POST", CHEFS_URL + '/api/v1/email', headers=ches_headers, data=ches_pay_load)
+      #_ches_content = json.loads(ches_response.content)
```
Could you please remove the commented out code.

rxjeong replied:
Removed the comment

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+EMAIL_SUBJECT_HIGH_VOLUME = config['EMAIL_TITLE']['EMAIL_SUBJECT_HIGH_VOLUME']
+
+
+REGIONAL_DISTRICT_NAME_DIC = dict(regionalDistrictOfBulkleyNechako='Regional District of Bulkley-Nechako'
```
Could we please move the dicts and arrays to separate constant files to reduce the lines of code on this file and have proper seggregation?

rxjeong replied:
Moved the dicts and arrays into constant.py

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+
+
+
+def convert_regional_district_to_name(id):
```
could we please rename the `id` to `key` for all the functions
https://stackoverflow.com/questions/77552/id-is-a-bad-variable-name-in-python

rxjeong replied:
Renamed to key for all the functions.

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+def create_receiving_site_lan_uses(cefs_dic, field):
+  _land_uses = []
+  for _k, _v in cefs_dic[field].items():
+    if _v == True:
```
should the `==` be replaced by `is`?

rxjeong replied:
I found this Python PE8 style guideline. So the comparing with Boolean was changed to following the guideline (e.g "if greeting:")

https://peps.python.org/pep-0008/#programming-recommendations

![image](https://user-images.githubusercontent.com/107894308/195915404-dd91e4ff-133b-4261-8a19-9a800910b96e.png)

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+      working_dic['totalSoilVolme'] = _total_soil_volume
+
+def create_site_relocation_email_msg(regional_district, popup_links):
+  msg = '<p>Soil Relocation Notifications are received by the ministry under section 55 of the <i>Environmental Management Act</i>. For more information on soil relocation from commercial and industrial sites in BC, please visit our <a href=https://soil-relocation-information-system-governmentofbc.hub.arcgis.com/>webpage</a>.</p>'
```
Could we make these as strings as templates put them into another file and read the templates from there?
https://www.geeksforgeeks.org/template-class-in-python/

rxjeong replied:
Changed to use email template files using Jinja Template module

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+def map_source_site(submission):
+  _src_dic = {}
+  _confirmation_id = helper.get_confirm_id(submission)
+  if (helper.validate_lat_lon(submission.get("A3-SourceSiteLatitude-Degrees"), submission.get("A3-SourceSiteLatitude-Minutes"), submission.get("A3-SourceSiteLatitude-Seconds"), 
```
could we put the hardcoded strings like `A3-SourceSiteLatitude-Degrees` into a constants file?

rxjeong replied:
Moved the hardcoded CHEFS parameter names into the constants.py

mishraomp commented on chefs_soils.py:
```diff
@@ -1,73 +1,1379 @@
+  print('Error: Source Site CSV Item ID is invalid!')
+else:
+  _srcCsvUpdateResult = _srcCsvItem.update({}, SOURCE_CSV_FILE)
+  print('Updated Soil Relocation Source Site CSV sucessfully: ' + str(_srcCsvUpdateResult))
```
A little typo around `sucessfully` should be `successfully` in multiple places.

rxjeong replied:
fixed the typo.

mishraomp reviewed (commented):
I have added review comments in individual files.
Could we have some kind of code formatting or linter for the entire application code base? something like pylint or similar.

rxjeong commented:
> I have added review comments in individual files. Could we have some kind of code formatting or linter for the entire application code base? something like pylint or similar.

I ran pylint and fixed tons of formatting issues, but I disabled the 'line-too-long' and 'no-member' for "get member in _gis.content" pylint message. I think the 'line-too-long' is not useful, but I put some changes to be readable for codes.
